File: src/main/java/com/kotlarz/backend/web/dto/TemperatureLogDto.java
package com.kotlarz.backend.web.dto;

import com.kotlarz.backend.domain.TemperatureLog;
import lombok.Builder;
import lombok.Data;

import java.util.Date;

@Data
@Builder
public class TemperatureLogDto {
    private Long id;

    private Date date;

    private Double value;

    public static TemperatureLogDto create(TemperatureLog temperatureLog) {
        return TemperatureLogDto.builder()
                .id(temperatureLog.getId())
                .date(temperatureLog.getDate())
                .value(temperatureLog.getValue())
                .build();
    }
}

File: src/main/java/com/kotlarz/service/dto/TemperatureLogDto.kt
package com.kotlarz.service.dto

import com.kotlarz.service.domain.TemperatureLogDomain
import java.util.*

fun fromDomain(domain: TemperatureLogDomain): TemperatureLogDto {
    return TemperatureLogDto(date = domain.date, address = domain.address, value = domain.value)
}

data class TemperatureLogDto(val date: Date,
                             val address: String,
                             val value: Double)
